src/main/java/org/opensearch/knn/indices/ModelGraveyard.java

```diff
@@ -151,27 +153,75 @@ public static NamedDiff readDiffFrom(StreamInput streamInput) throws IOException
      * @throws IOException
      */
     public static ModelGraveyard fromXContent(XContentParser xContentParser) throws IOException {
+        // Added validation checks to validate all the different possible scenarios
+        // model_ids:"abcd" - Throws exception as the START_OBJECT token is missing
+        // {} - Returns an empty ModelGraveyard object (BackwardCompatibility)
+        // {["abcd", "1234"]} - Throws exception as the FIELD_NAME token is missing
+        // {"dummy_field_name":} - Throws exception as the FIELD_NAME is not matching with model_ids
+        // {model_ids:"abcd"} - Throws exception as the START_ARRAY token is missing after field name
+        // {model_ids:null} - Throws exception as the START_ARRAY token is missing
+        // {model_ids:[]} - Parses and returns an empty ModelGraveyard object as there are no model ids
+        // {model_ids: ["abcd", "1234"]} - Parses and returns a ModelGraveyard object which contains the model ids "abcd" and "1234"
+        // {model_ids:[],dummy_field:[]} - Throws exception as we have FIELD_NAME(dummy_field) instead of END_OBJECT token
+
         ModelGraveyard modelGraveyard = new ModelGraveyard();
 
         // If it is a fresh parser, move to the first token
         if (xContentParser.currentToken() == null) {
             xContentParser.nextToken();
         }
 
-        // on a start object move to next token
-        if (xContentParser.currentToken() == XContentParser.Token.START_OBJECT) {
-            xContentParser.nextToken();
+        // Validate if the first token is START_OBJECT
```

Comment:
nit: No need to add comment for each if condition if the if condition is readable

Reply:
I think it better to add comments as we might forget what the conditions are for.

Reply:
But it seems obvious from the code itself in this case.